src/main/java/com/pusher/client/channel/impl/ChannelManager.java

```diff
@@ -89,6 +98,12 @@ public void unsubscribeFrom(final String channelName) {
             return;
         }
         if (connection.getState() == ConnectionState.CONNECTED) {
+
+            // only private encrypted channels need to be securely torn down
+            if (channel instanceof PrivateEncryptedChannelImpl) {
```

Comment:
this change isn't nested inside a ``factory.queueOnEventThread(new Runnable() {` which i think is okay because we want to remove all the keys asap, and doesn't have any dependencies on anything else happening.